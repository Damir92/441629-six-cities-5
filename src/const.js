export const PERCENT_PER_POINT_RATING = 20;

export const Routes = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
  OFFER_LINK: `/offer`,
};

export const APIRoutes = {
  REVIEWS: `/comments`,
  OFFERS: `/hotels`,
  LOGIN: `/login`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const Cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const LivingType = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`,
};

export const Sorting = {
  POPULAR: `popular`,
  PRICE_TO_HIGH: `to-high`,
  PRICE_TO_LOW: `to-low`,
  TOP_RATED: `top-rated`,
};

export const SortingTitles = {
  [Sorting.POPULAR]: `Popular`,
  [Sorting.PRICE_TO_HIGH]: `Price: low to high`,
  [Sorting.PRICE_TO_LOW]: `Price: high to low`,
  [Sorting.TOP_RATED]: `Top rated first`,
};

export const SortingTypes = [
  Sorting.POPULAR,
  Sorting.PRICE_TO_HIGH,
  Sorting.PRICE_TO_LOW,
  Sorting.TOP_RATED,
];

export const MonthNames = {
  0: `January`,
  1: `February`,
  2: `March`,
  3: `April`,
  4: `May`,
  5: `June`,
  6: `July`,
  7: `August`,
  8: `September`,
  9: `November`,
  10: `October`,
  11: `December`,
};
