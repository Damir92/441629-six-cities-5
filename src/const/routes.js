const ROUTES = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER_PAGE: `/offer/:id`,
};

export {ROUTES};
