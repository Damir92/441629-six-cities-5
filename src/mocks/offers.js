export const offers = [
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!',
    'goods': [
      'Fridge',
      'Washer',
      'Coffee machine',
      'Washing machine',
      'Dishwasher',
      'Laptop friendly workspace',
      'Cable TV',
      'Air conditioning',
      'Baby seat',
      'Towels',
      'Breakfast',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 2,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37454,
      'longitude': 4.881976,
      'zoom': 16,
    },
    'volume': 1,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    'price': 111,
    'rating': 3.3,
    'title': 'Wood and stone place',
    'type': 'room',
  },
  {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.',
    'goods': [ 'Laptop friendly workspace', 'Baby seat', 'Towels', 'Air conditioning', 'Breakfast', 'Washer' ],
    'host': { 'avatar': 'img/avatar-angelina.jpg', 'id': 25, 'isPro': true, 'name': 'Angelina' },
    'id': 4,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': { 'latitude': 52.35054, 'longitude': 4.908976, 'zoom': 16 },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
    'price': 294,
    'rating': 3.8,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 4,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Towels',
      'Breakfast',
      'Washer',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 9,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.37854,
      'longitude': 4.894976,
      'zoom': 16,
    },
    'volume': 10,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
    'price': 406,
    'rating': 2.2,
    'title': 'Loft Studio in the Central Area',
    'type': 'hotel',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.',
    'goods': [
      'Washer',
      'Towels',
      'Breakfast',
      'Air conditioning',
      'Baby seat',
      'Fridge',
      'Laptop friendly workspace',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 19,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.397540000000006,
      'longitude': 4.9099759999999995,
      'zoom': 16,
    },
    'volume': 2,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
    'price': 286,
    'rating': 2.8,
    'title': 'The Joshua Tree House',
    'type': 'room',
  }, {
    'bedrooms': 1,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    'goods': [
      'Coffee machine',
      'Breakfast',
      'Laptop friendly workspace',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Dishwasher',
      'Washer',
      'Fridge',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25,
      'isPro': true,
      'name': 'Angelina',
    },
    'id': 22,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
    ],
    'isFavorite': false,
    'isPremium': true,
    'location': {
      'latitude': 52.37554,
      'longitude': 4.9019759999999994,
      'zoom': 16,
    },
    'volume': 5,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 143,
    'rating': 4.6,
    'title': 'Canal View Prinsengracht',
    'type': 'house',
  }, {
    'bedrooms': 2,
    'city': {
      'name': 'Amsterdam',
      'location': {
        'latitude': 52.37454,
        'longitude': 4.897976,
        'zoom': 13,
      },
    },
    'description': 'Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.',
    'goods': [
      'Washer',
      'Fridge',
      'Air conditioning',
      'Towels',
      'Baby seat',
      'Breakfast',
      'Laptop friendly workspace',
      'Dishwasher',
      'Coffee machine',
    ],
    'host': {
      'avatar': 'img/avatar-angelina.jpg',
      'id': 25, 'isPro': true,
      'name': 'Angelina',
    },
    'id': 24,
    'images': [
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg',
      'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg',
    ],
    'isFavorite': false,
    'isPremium': false,
    'location': {
      'latitude': 52.367540000000005,
      'longitude': 4.883976,
      'zoom': 16,
    },
    'volume': 9,
    'preview': 'https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg',
    'price': 289,
    'rating': 2,
    'title': 'House in countryside',
    'type': 'house',
  },
];
