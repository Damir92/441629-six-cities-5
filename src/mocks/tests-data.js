export const cityOffers = [
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    goods: [
      `Fridge`,
      `Washer`,
      `Coffee machine`,
      `Washing machine`,
      `Dishwasher`,
      `Laptop friendly workspace`,
      `Cable TV`,
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Breakfast`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37454,
      longitude: 4.881976,
      zoom: 16,
    },
    volume: 1,
    preview: `previewURL`,
    price: 111,
    rating: 3.3,
    title: `Wood and stone place`,
    type: `room`,
  },
  {
    bedrooms: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    goods: [
      `Laptop friendly workspace`,
      `Baby seat`,
      `Towels`,
      `Air conditioning`,
      `Breakfast`,
      `Washer`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 4,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 52.35054,
      longitude: 4.908976,
      zoom: 16,
    },
    volume: 2,
    preview: `previewURL`,
    price: 294,
    rating: 3.8,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  }, {
    bedrooms: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a "slow" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Fridge`,
      `Towels`,
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`,
    ],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    id: 9,
    images: [
      `imgURL-1`,
      `imgURL-2`,
    ],
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.37854,
      longitude: 4.894976,
      zoom: 16,
    },
    volume: 10,
    preview: `previewURL`,
    price: 406,
    rating: 2.2,
    title: `Loft Studio in the Central Area`,
    type: `hotel`,
  },
];
