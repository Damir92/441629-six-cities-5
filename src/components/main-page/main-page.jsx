import React from 'react';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {changeCityAction, setSortingTypeAction} from '../../store/action';
import {getCity, getSortedOffers} from '../../store/reducers/site-data/selectors';
import {getAuthorizationStatus, getUserData} from '../../store/reducers/user/selectors';

import {offerPropTypes} from '../../prop-types';
import {AuthorizationStatus, Cities, Routes} from '../../const';

import OffersList from '../offers-list/offers-list';
import OffersMap from '../offers-map/offers-map';
import CitiesList from '../cities-list/cities-list';
import OffersSorting from '../offers-sorting/offers-sorting';

const MainPage = ({city, cityOffers = [], logged, userData, onCityClick, onOptionClick}) => {
  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </a>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <Link
                    className="header__nav-link header__nav-link--profile"
                    to={logged === AuthorizationStatus.AUTH
                      ?
                      Routes.FAVORITES
                      :
                      Routes.LOGIN
                    }
                  >
                    <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                    {logged === AuthorizationStatus.AUTH
                      ?
                      <span className="header__user-name user__name">
                        {userData.email}
                      </span>
                      :
                      <span className="header__login">Sign in</span>
                    }
                  </Link>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">

            <CitiesList
              city={city}
              onCityClick={onCityClick}
            />

          </section>
        </div>
        <div className="cities">
          <div className={`cities__places-container container ${cityOffers.length === 0 ? `cities__places-container--empty` : ``}`}>

            {cityOffers.length === 0
              ?
              <section className="cities__no-places">
                <div className="cities__status-wrapper tabs__content">
                  <b className="cities__status">No places to stay available</b>
                  <p className="cities__status-description">We could not find any property available at the moment in {city}</p>
                </div>
              </section>
              :
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">{cityOffers.length} places to stay in {city}</b>

                <OffersSorting
                  onOptionClick={onOptionClick}
                />

                <OffersList
                  cityOffers={cityOffers}
                />

              </section>
            }

            <div className="cities__right-section">
              <section className="cities__map map">
                <OffersMap
                  cityOffers={cityOffers}
                />
              </section>
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

MainPage.propTypes = {
  city: PropTypes.oneOf(Cities).isRequired,
  cityOffers: PropTypes.arrayOf(
      PropTypes.shape(offerPropTypes).isRequired
  ).isRequired,
  logged: PropTypes.oneOf([AuthorizationStatus.AUTH, AuthorizationStatus.NO_AUTH]).isRequired,
  userData: PropTypes.oneOf([PropTypes.shape({
    email: PropTypes.string,
  }), ``]),
  onCityClick: PropTypes.func.isRequired,
  onOptionClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  cityOffers: getSortedOffers(state),
  logged: getAuthorizationStatus(state),
  userData: getUserData(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (city) => dispatch(changeCityAction(city)),
  onOptionClick: (type) => dispatch(setSortingTypeAction(type)),
});

export {MainPage};
export default connect(mapStateToProps, mapDispatchToProps)(MainPage);
