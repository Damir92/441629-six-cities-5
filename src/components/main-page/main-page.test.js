import React from 'react';
import {shallow} from 'enzyme';

import {cityOffers, noop} from '../../mocks/tests-data';
import {Cities, AuthorizationStatus} from '../../const';

import {MainPage} from './main-page';

const history = {
  push: noop,
};

describe(`MainPage renders correctly`, () => {
  it(`MainPage renders correctly`, () => {
    const component = shallow(
        <MainPage
          city={Cities[0]}
          cityOffers={cityOffers}
          history={history}
          logged={AuthorizationStatus.NO_AUTH}
          userData={null}
          onCityClick={noop}
          onOptionClick={noop}
        />
    );

    expect(component).toMatchSnapshot();
  });
});
