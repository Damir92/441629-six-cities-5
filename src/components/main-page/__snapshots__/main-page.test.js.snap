// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly MainPage renders correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CitiesList
          city="Paris"
          onCityClick={[Function]}
        />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container "
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Paris
          </b>
          <Connect(WithToggle)
            onOptionClick={[Function]}
          />
          <OffersList
            cityOffers={
              Array [
                Object {
                  "bedrooms": 1,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.37454,
                      "longitude": 4.897976,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Fridge",
                    "Washer",
                    "Coffee machine",
                    "Washing machine",
                    "Dishwasher",
                    "Laptop friendly workspace",
                    "Cable TV",
                    "Air conditioning",
                    "Baby seat",
                    "Towels",
                    "Breakfast",
                  ],
                  "host": Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 2,
                  "images": Array [
                    "imgURL-1",
                    "imgURL-2",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.881976,
                    "zoom": 16,
                  },
                  "preview": "previewURL",
                  "price": 111,
                  "rating": 3.3,
                  "title": "Wood and stone place",
                  "type": "room",
                  "volume": 1,
                },
                Object {
                  "bedrooms": 1,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.37454,
                      "longitude": 4.897976,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.",
                  "goods": Array [
                    "Laptop friendly workspace",
                    "Baby seat",
                    "Towels",
                    "Air conditioning",
                    "Breakfast",
                    "Washer",
                  ],
                  "host": Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 4,
                  "images": Array [
                    "imgURL-1",
                    "imgURL-2",
                  ],
                  "isFavorite": true,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 52.35054,
                    "longitude": 4.908976,
                    "zoom": 16,
                  },
                  "preview": "previewURL",
                  "price": 294,
                  "rating": 3.8,
                  "title": "Loft Studio in the Central Area",
                  "type": "hotel",
                  "volume": 2,
                },
                Object {
                  "bedrooms": 4,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.37454,
                      "longitude": 4.897976,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \\"slow\\" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                  "goods": Array [
                    "Air conditioning",
                    "Baby seat",
                    "Fridge",
                    "Towels",
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 9,
                  "images": Array [
                    "imgURL-1",
                    "imgURL-2",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 52.37854,
                    "longitude": 4.894976,
                    "zoom": 16,
                  },
                  "preview": "previewURL",
                  "price": 406,
                  "rating": 2.2,
                  "title": "Loft Studio in the Central Area",
                  "type": "hotel",
                  "volume": 10,
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Connect(OffersMap)
              cityOffers={
                Array [
                  Object {
                    "bedrooms": 1,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.37454,
                        "longitude": 4.897976,
                        "zoom": 13,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                    "goods": Array [
                      "Fridge",
                      "Washer",
                      "Coffee machine",
                      "Washing machine",
                      "Dishwasher",
                      "Laptop friendly workspace",
                      "Cable TV",
                      "Air conditioning",
                      "Baby seat",
                      "Towels",
                      "Breakfast",
                    ],
                    "host": Object {
                      "avatar": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 2,
                    "images": Array [
                      "imgURL-1",
                      "imgURL-2",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.37454,
                      "longitude": 4.881976,
                      "zoom": 16,
                    },
                    "preview": "previewURL",
                    "price": 111,
                    "rating": 3.3,
                    "title": "Wood and stone place",
                    "type": "room",
                    "volume": 1,
                  },
                  Object {
                    "bedrooms": 1,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.37454,
                        "longitude": 4.897976,
                        "zoom": 13,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.",
                    "goods": Array [
                      "Laptop friendly workspace",
                      "Baby seat",
                      "Towels",
                      "Air conditioning",
                      "Breakfast",
                      "Washer",
                    ],
                    "host": Object {
                      "avatar": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 4,
                    "images": Array [
                      "imgURL-1",
                      "imgURL-2",
                    ],
                    "isFavorite": true,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.35054,
                      "longitude": 4.908976,
                      "zoom": 16,
                    },
                    "preview": "previewURL",
                    "price": 294,
                    "rating": 3.8,
                    "title": "Loft Studio in the Central Area",
                    "type": "hotel",
                    "volume": 2,
                  },
                  Object {
                    "bedrooms": 4,
                    "city": Object {
                      "location": Object {
                        "latitude": 52.37454,
                        "longitude": 4.897976,
                        "zoom": 13,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \\"slow\\" pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                    "goods": Array [
                      "Air conditioning",
                      "Baby seat",
                      "Fridge",
                      "Towels",
                      "Breakfast",
                      "Washer",
                      "Laptop friendly workspace",
                    ],
                    "host": Object {
                      "avatar": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 9,
                    "images": Array [
                      "imgURL-1",
                      "imgURL-2",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 52.37854,
                      "longitude": 4.894976,
                      "zoom": 16,
                    },
                    "preview": "previewURL",
                    "price": 406,
                    "rating": 2.2,
                    "title": "Loft Studio in the Central Area",
                    "type": "hotel",
                    "volume": 10,
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
